### PR TITLE
Make Aarch64 not a legacy platform

### PR DESCRIPTION
**Pull request by annop-w**
This allows for MKL convolution primitives to always be cached on Aarch64.

### CODE REVIEW

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/util/mkl_util.h
Comment (annop-w) [reply tensorflow/core/util/mkl_util.h]: Done. Thank you.

**Review by penpornk — APPROVED** [review]

File: tensorflow/core/util/mkl_util.h
Comment (annop-w) [reply tensorflow/core/util/mkl_util.h]: Done.

**Review by penpornk — APPROVED** [review]
Thank you again for the changes!

**Review by penpornk — APPROVED** [review]
Not sure why this hasn't been merged. Reapproving again.

**Comment by penpornk** [comment]
@annop-w Could you please help a take a look at the [Arm CI failure](https://github.com/tensorflow/tensorflow/actions/runs/3938039126)? Here is the [log](https://github.com/tensorflow/tensorflow/actions/runs/3938039126/jobs/6736273388) (or click here for [raw log file](https://pipelines.actions.githubusercontent.com/serviceHosts/a4c1c801-a9a8-4666-a73a-996f5c2479f7/_apis/pipelines/1/runs/57761/signedlogcontent/2?urlExpires=2023-01-17T14%3A12%3A33.8719264Z&urlSigningMethod=HMACV1&urlSignature=RBH2%2BfILPlzMOAq6kFQxtZ4Xx7sMtFqfdYiXG2SkjUM%3D)).

**Comment by annop-w** [comment]
@penpornk Thanks for pointing that out. I'm investigating.

**Comment by annop-w** [comment]
@penpornk I have been trying to reproduce the failure locally, but without success so far. Could we try running the CI pipeline once more ?

**Comment by penpornk** [comment]
@annop-w Sure! I've triggered another run. You can see the job [here](https://github.com/tensorflow/tensorflow/actions/runs/3988056012/jobs/6839402805).

**Comment by annop-w** [comment]
The ARM CI pipeline failed even before running the tests https://github.com/tensorflow/tensorflow/actions/runs/3988056012/jobs/6839402805#step:5:16929

**Comment by penpornk** [comment]
@annop-w That's weird. I just triggered another rerun.

**Comment by annop-w** [comment]
@penpornk Looks like the pipeline is broken. All red for other PRs with the same error.

**Comment by penpornk** [comment]
@annop-w Sorry about that. I'm checking with @elfringham on the pipeline.

**Comment by elfringham** [comment]
Pipeline fix in https://github.com/tensorflow/tensorflow/pull/59430

**Review by penpornk — APPROVED** [review]
Thank you for rebasing! Let's rerun the tests.

**Comment by penpornk** [comment]
@annop-w The ARM CI got [23 test failures](https://github.com/tensorflow/tensorflow/actions/runs/4007933935/jobs/6881448921). The job before it only had [7 tests failed](https://github.com/tensorflow/tensorflow/actions/runs/4007920238/jobs/6881416173) so I think the rest is specific to this PR.

**Comment by annop-w** [comment]
@penpornk Yes, I am able to reproduce the failures now. There is probably a bug on the oneDNN side, which got revealed after the `MklConvBwdInputPrimitive` is cached. I am still investigating. Thanks for pointing out.

**Review by penpornk — APPROVED** [review]
Thank you for the fixes! I'll probably wait until after the branch cut to merge this.

**Pull request merged**
